src/Features/CSharp/Portable/ImplementInterface/CSharpImplementInterfaceService.cs

```diff
@@ -120,28 +108,32 @@ protected override Document ImplementDisposePattern(Document document, SyntaxNod
     //   Dispose(false);
     // }}
 
-    // {CSharpFeaturesResources.This_code_added_to_correctly_implement_the_disposable_pattern}
     {(explicitly ? "void System.IDisposable." : "public void ")}Dispose()
     {{
         // {CSharpFeaturesResources.Do_not_change_this_code_Put_cleanup_code_in_Dispose_bool_disposing_above}
-        Dispose(true);
+        Dispose(true);";
+
+            if (classSymbol.IsSealed)
```

Comment:
@sharwell to review the changes here - he has been pushing for much wider changes to the Disposable pattern code.

Reply:
> @sharwell to review the changes here - he has been pushing for much wider changes to the Disposable pattern code.

That's fine.  But i would very much like to have this happen a step at a time.  We can already improve a bunch of things.  So we should get in what we can, and continue to improve later if more tweaks are desired.

Reply:
I'm fine with direct simplification of the current approach (even though I believe it's unnecessarily verbose). However, I don't believe the `IsSealed` distinction is beneficial. We should just generate the `SuppressFinalize` call in all cases.

Reply:
> I'm fine with direct simplification of the current approach (even though I believe it's unnecessarily verbose). However, I don't believe the IsSealed distinction is beneficial. We should just generate the SuppressFinalize call in all case

Sounds good! I'll switch to that.  I don't liek that either.  :)